File: Semester 4/Class11_Computer_Contest_Level_3_Notes_and_Homework_Nov_19/lvl3_hw_sol/time_on_task.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int t, c, i, total=0, time[101];
    cin >> t >> c;
    for(i=0; i<c; i++)
        cin >> time[i];
    sort(time, time+c);
    for(i=0; i<c; i++)
        if(total+time[i]<=t)
            total += time[i];
        else{
            cout << i << endl;
            return 0;
        }
    cout << i+1 << endl;
    return 0;
}

File: Semester 4/Class11_Computer_Contest_Level_3_Notes_and_Homework_Nov_19/lvl3_hw_sol/time_on_task.java
import java.util.Arrays;
import java.util.Scanner;

public class ccc13j4 {
	
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int time = s.nextInt();
		int cnum = s.nextInt();
		int[] ctimes = new int[100];
		for (int i = 0; i < cnum; i++)
			ctimes[i] = s.nextInt();
		Arrays.sort(ctimes, 0, cnum);
		int counter = 0;
		for (int i = 0; i < cnum && time >= 0; i++) {
			time -= ctimes[i];
			if (time < 0) break;
			else counter++;
		}
		System.out.println(counter);
		s.close();
	}
	
}
